### PR TITLE
Remove RangeConstraints in favor of ValueRanges

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110757
* __->__ #109859

cc @avikchaudhuri @gmagogsfm @zhxchen17

### CODE REVIEW

File: torch/_export/exported_program.py [hunk 1]
```diff
@@ -334,29 +334,29 @@ def _process_constraints(
                     equality_constraints.append((node_dim, other_node_dim))
 
     # Create dict mapping symbol to a singular range (lower, upper)
-    range_constraints: Dict[sympy.Symbol, RangeConstraint] = {}
+    range_constraints: Dict[sympy.Symbol, ValueRanges] = {}
 
     # Add inline constraints to range_constraints
     for symbol, value_range in inline_constraints.items():
-        range_constraints[symbol] = RangeConstraint(value_range.lower, value_range.upper)
+        range_constraints[symbol] = ValueRanges(value_range.lower, value_range.upper)
```
Comment (avikchaudhuri) [inline torch/_export/exported_program.py:341]: pretty sure `value_range`is already of type `ValueRanges`. In which case you can initialize `range_constraints = {**inline_constraints}`

File: torch/_export/exported_program.py [hunk 2]
```diff
@@ -334,29 +334,29 @@ def _process_constraints(
...
+        assert min_val <= max_val  # type: ignore[operator]
 
         # Add input node range constraints
         val = placeholder_nodes[input_dim.input_name].meta["val"]
         assert isinstance(val, FakeTensor)
         symint = val.shape[input_dim.dim]
         assert isinstance(symint, SymInt), f"Expected SymInt but got {symint}: {type(symint)}"
         symbol = symint.node._expr
-        range_constraints[symbol] = RangeConstraint(min_val, max_val)
+        range_constraints[symbol] = ValueRanges(min_val, max_val)
```
Comment (avikchaudhuri) [inline torch/_export/exported_program.py:359]: See if you can simply use `&` on `ValueRanges`, which would be a much better way of doing this and also take advantage of this change.

**Review by avikchaudhuri — APPROVED** [review]
nits

File: torch/_export/passes/add_runtime_assertions_for_constraints_pass.py [hunk 3]
```diff
@@ -14,22 +13,17 @@
 from torch.fx.experimental.symbolic_shapes import SymInt
 from torch._export.pass_base import _ExportPassBase, ProxyValue, PassResult
 from torch._subclasses.fake_tensor import FakeTensor
+from torch.utils._sympy.value_ranges import ValueRanges
 
 
-__all__ = ["_AddRuntimeAssertionsForConstraintsPass", "InputDim", "RangeConstraint"]
```
Comment (kit1980) [inline torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:19]: Don't remove public API without a deprecation period.

**Comment by kit1980** [comment]
I'm reverting the stack in https://github.com/pytorch/pytorch/pull/111115

**Comment by kit1980** [comment]
@tugsbayasgalan did an internal forward-fix instead of revert.
